host_provider/providers/azure.py

```diff
@@ -0,0 +1,342 @@
+                           subscription_id=self.credential.subscription_id)
+
+    def get_node(self, node_id):
+        pass
+
+    def _parse_image(self, name, size, gallery='myGallery', image='mssql_2019_0_0', version='1.0.0'):
+        sql_dict = OrderedDict()
+        templates = JsonTemplates()
+        pw = self.credential.init_password
+        region = self.credential.region
+
+        image_id = '/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/galleries/%s/images/%s/versions/%s' \
+                   % (self.credential.subscription_id, self.credential.resource_group, gallery, image, version)
+
+        network_id = '/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/networkInterfaces/%s' \
+                     % (self.credential.subscription_id, self.credential.resource_group, name)
+
+        osProfile = {'adminUsername': 'dbaas', 'computerName': name, 'adminPassword': pw}
+
+        try:
+            for file in templates.list_files(version):
+                file_name = file.name.split('.json')[0]
+                if file_name == 'sql':
+                    sql_dict = templates.load_json(file.as_posix())
+                    sql_dict['properties']['hardwareProfile']['vmSize'] = size
+                    sql_dict['properties']['storageProfile']['imageReference']['id'] = image_id
+                    sql_dict['properties']['storageProfile']['osDisk']['name'] = name
+                    sql_dict['properties']['osProfile'] = osProfile
+                    sql_dict['properties']['networkProfile']['networkInterfaces'][0]['id'] = network_id
+                    sql_dict['location'] = region
+            return sql_dict
+        except Exception as error:
+            raise Exception("Template parse error: {}" % error)
+
+    def offering_to(self, cpu, memory, api_version='2020-12-01'):
+
+        base_url = self.credential.endpoint
+        action = "subscriptions/%s/providers/Microsoft.Compute/locations/%s/vmSizes?api-version=%s" \
+                 % (self.credential.subscription_id, self.credential.region, api_version)
+
+        header = {}
+        self.get_azure_connection()
+        self.azClient.connect(base_url=base_url)
+        self.azClient.add_default_headers(header)
+        self.azClient.connection.request("GET", action, headers=header)
+        resp = self.azClient.connection.getresponse()
+
+        if resp.status_code == 200:
+            offerings = resp.json()['value']
+
+            for offering in offerings:
+                if offering.get('memoryInMB') == memory and offering.get('numberOfCores') == cpu:
+                    return offering
+
+        raise OfferingNotFoundError(
+            "Offering with {} cpu and {} of memory not found.".format(cpu, memory)
+        )
+
+    def _parse_nic(self, name, vnet, subnet, version='1.0.0'):
+        nic_dict = OrderedDict()
+        id = '/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/virtualNetworks/%s/subnets/%s' \
```

Comment:
- Hardcoded url.
- use format instead of %
- do no use \ to break lines, use parentesis